annotations/views/repository_views.py

```diff
@@ -209,23 +209,34 @@ def repository_list(request):
 @citesphere_authenticated
 def repository_details(request, repository_id):
     template = "annotations/repository_details.html"
-    user = None if isinstance(request.user, AnonymousUser) else request.user
     repository = get_object_or_404(Repository, pk=repository_id)
+    user = None if isinstance(request.user, AnonymousUser) else request.user
+    project_id = request.GET.get('project_id')
+
+    # Fetch text collections (projects) owned by the current user
+    user_owned_collections = TextCollection.objects.filter(ownedBy=user)
+    texts_by_project = {}
+
+    for collection in user_owned_collections:
+        # Filter texts within each collection that belong to the repository
+        collection_texts = collection.texts.filter(repository=repository).order_by('-added')
+        if collection_texts.exists():
+            texts_by_project[collection] = list(collection_texts)
```

Comment:
does this mean that all texts in a collection will be loaded here? Those should be paginated, since they might be a lot.